spinning-button.js

```diff
@@ -1,51 +1,52 @@
-import React, { Component } from 'react'
+import React, {Component} from 'react'
+import * as t from 'prop-types'
 
-import {
-  StyleSheet,
-  View,
-  Animated,
-  TouchableOpacity
-} from 'react-native'
+import {Animated, TouchableOpacity} from 'react-native'
 
-export default class SpinningButton extends Component {
+class SpinningButton extends Component {
   constructor(props) {
     super(props)
     this.state = {
-      rotateValue: new Animated.Value(0)
+      rotateValue: new Animated.Value(0),
+      active: true
     }
   }
 
   _onPressButton = () => {
-    if (this.state.rotateValue._value == 0 || (this.state.rotateValue._value % 360) == 0) {
-      Animated.spring(
-        this.state.rotateValue,
-        {toValue: this.state.rotateValue._value + 360,
-        }).start()
+    if (
+      this.state.rotateValue._value === 0 ||
+      this.state.rotateValue._value % 360 === 0
+    ) {
+      Animated.spring(this.state.rotateValue, {
+        toValue: this.state.rotateValue._value + 360
+      }).start(() => this.props.onPress())
     }
-
-    this.props.onPress();
```

Comment:
Nice! I guess we may take this line out now =)
//this.props.onPress();

Reply:
The lines with the "-" in front of them are the removals of git, so yes it is gonne ;)